Detectors/TPC/reconstruction/src/KrBoxClusterFinder.cxx

```diff
@@ -0,0 +1,250 @@
+// Copyright CERN and copyright holders of ALICE O2. This software is
+// distributed under the terms of the GNU General Public License v3 (GPL
+// Version 3), copied verbatim in the file "COPYING".
+//
+// See http://alice-o2.web.cern.ch/license for full licensing information.
+//
+// In applying this license CERN does not waive the privileges and immunities
+// granted to it by virtue of its status as an Intergovernmental Organization
+// or submit itself to any jurisdiction.
+
+/// \file KrBoxClusterFinder3D.cpp
+/// \brief Class source code for Krypton and X-ray events
+/// \author Philip Hauer <hauer@hiskp.uni-bonn.de>
+
+#include "TPCReconstruction/KrBoxClusterFinder.h"
+#include <iostream>
+
+using namespace o2::tpc;
+
+// Constructor:
+// This function creates a three dimensional vector (Pad,Row,Time) which is
+// filled with the charges
+KrBoxClusterFinder::KrBoxClusterFinder(std::vector<o2::tpc::Digit>* eventSector)
+{
+  std::vector<o2::tpc::Digit>::iterator cl = eventSector->begin();

```

Comment:
This can be removed if you use range based for below.

Reply:
I need "i" for the next lines.